packages/transformers/webext/src/WebExtTransformer.js

```diff
@@ -0,0 +1,261 @@
+// @flow
+
+import type {MutableAsset} from '@parcel/types';
+import {Transformer} from '@parcel/plugin';
+import {join, extname, dirname, relative} from 'path';
+import {parse} from 'json-source-map';
+import parseCSP from 'content-security-policy-parser';
+import {validateSchema} from '@parcel/utils';
+import ThrowableDiagnostic, {getJSONSourceLocation} from '@parcel/diagnostic';
+import {glob} from '@parcel/utils';
+import type {DiagnosticCodeHighlight} from '@parcel/diagnostic';
+import WebExtSchema from './schema';
+
+const BASE_KEYS = ['manifest_version', 'name', 'version'];
+
+const DEP_LOCS = [
+  ['icons'],
+  ['browser_action', 'default_icon'],
+  ['browser_action', 'default_popup', true],
+  ['browser_action', 'theme_actions', 'light'],
+  ['browser_action', 'theme_actions', 'dark'],
+  ['page_action', 'default_icon'],
+  ['page_action', 'default_popup', true],
+  ['background', 'scripts', true],
+  ['chrome_url_overrides'],
+  ['devtools_page', true],
+  ['options_ui', 'page', true],
+  ['sidebar_action', 'default_icon'],
+  ['sidebar_action', 'default_panel', true],
+  ['storage', 'managed_schema'],
+  ['theme', 'images', 'theme_frame'],
+  ['theme', 'images', 'additional_backgrounds'],
+  ['user_scripts', 'api_script', true],
+];
+
+async function collectDependencies(
+  asset: MutableAsset,
+  program: any,
+  ptrs: {[key: string]: any, ...},
+) {
+  // isEntry used whenever strictly necessary to preserve filename
+  // also for globs because it's wasteful to write out every file name
+  const fs = asset.fs;
+  const filePath = asset.filePath;
+  if (program.default_locale) {
+    const locales = join(dirname(filePath), '_locales');
+    let err = !(await fs.exists(locales))
+      ? 'key'
+      : !(await fs.exists(join(locales, program.default_locale)))
+      ? 'value'
+      : null;
+    if (err) {
+      throw new ThrowableDiagnostic({
+        diagnostic: [
+          {
+            message:
+              'Localization directory' +
+              (err == 'value' ? ' for ' + program.default_locale : '') +
+              ' does not exist',
+            origin: '@parcel/transformer-webext',
+            filePath,
+            codeFrame: {
+              codeHighlights: [
+                ((getJSONSourceLocation(
+                  ptrs['/default_locale'],
+                  err,
+                ): any): DiagnosticCodeHighlight),
+              ],
+            },
+          },
+        ],
+      });
```

Comment:
1. The error doesn't say where the locale folder should be
2. Don't just cast a flow error away: add your current message to the highlight and change the other message to something more top-level:

```js
      throw new ThrowableDiagnostic({
        diagnostic: [
          {
            message: 'Failed to process WebExtension manifest',
            filePath,
            codeFrame: {
              codeHighlights: [
                {
                  ...getJSONSourceLocation(ptrs['/default_locale'], err),
                  message: `Localization directory${
                    err == 'value' ? ' for ' + program.default_locale : ''
                  } does not exist: ${relative(
                    dirname(filePath),
                    join(locales, program.default_locale),
                  )}`,
                },
              ],
            },
          },
        ],
      });
```

<img width="707" alt="Bildschirmfoto 2020-11-14 um 23 29 24" src="https://user-images.githubusercontent.com/4586894/99158257-3be7cf00-26d1-11eb-8bfc-9c60e30d4710.png">